code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -136,6 +136,9 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 	//Copies variables from sample
 	//Returns: 1 if we are mutable, 0 otherwise
 
+/datum/gas_mixture/proc/ratio_copy_from(datum/gas_mixture/sample, ratio = 1)

```

Comment:
Are you just declaring a proc before defining it like in C++?

Reply:
That's the style of this file so I stuck to form